### DISCUSSION TITLE
Multiple Instances of the Same Track When Using simplemaxtracks

### DISCUSSION THREAD

**Discussion by mollywestbrook** [talmolab/sleap#1798] [Help!]
Hi SLEAP devs and community! I'm using SLEAP 1.3.3 installed on a remote HPCC, and have transitioned to entirely training and tracking videos via command line. We use a top down model, with the most recent iteration containing 1914 frames across 145 videos in its training dataset. Generally, the outputs are fine, but I wanted to alert you to an ongoing issue that may be a bug. I use the following flags on our track script:

`--tracking.tracker simplemaxtracks --tracking.max_tracking 1 --tracking.max_tracks 5 --tracking.target_instance_count 5 --tracking.pre_cull_to_target 1 --tracking.pre_cull_iou_threshold 0.8 --tracking.similarity iou --tracking.match hungarian --tracking.track_window 5 --tracking.save_shifted_instances 0 --tracking.kf_init_frame_count 10 --tracking.kf_node_indices 1,2,3,4 --tracking.post_connect_single_breaks 1`

According to docs and discussions here, this should result in a hard cap of 5 tracks. However, we're noticing that we receive multiple instances of the same track, demonstrated below: 
![image](https://github.com/talmolab/sleap/assets/48096260/faad2005-8ed2-47f8-b56f-29cff4dfe7b8)

Also, there will be several frames where an instance will appear in our dataset, but the instance is unlabelled, and labeling it will not propagate all instances of the unlabelled track:
![image](https://github.com/talmolab/sleap/assets/48096260/f91857e6-889d-4318-be9b-5ce6fb21edde)

Checking the h5 on this one we've got 13 instances, so tracks aren't getting merged during export. Any guidance would be helpful on this one! Not sure if it's a bug or not, if it's worth updating or anything like that. Hope the lab is doing well :)

**Comment by talmo** [comment]
Hi @mollywestbrook,

Thanks for the report! This should probably not be happening. Do you mind trying with a simplified set of flags:

```
--tracking.tracker simplemaxtracks --tracking.max_tracking 1 --tracking.max_tracks 5 --tracking.pre_cull_iou_threshold 0.8 --tracking.similarity iou --tracking.match hungarian --tracking.track_window 5
```

Let me know if that still leads to >5 tracks. We'll debug from there.

Thanks!

Talmo

**Reply by mollywestbrook** [reply]
I will test these out on the same video. Updates to come after lunch.

**Reply by mollywestbrook** [reply]
Update: got the data back after an exhausting wait in the queue and it looks great, exactly five tracks: 

![image](https://github.com/talmolab/sleap/assets/48096260/76c3ce37-b12a-4ff2-bde6-b41540e318d6)

Double checked the h5 and we also have 5 instances. We'll plan to use the simplified set of parameters for now, but I did have one of our undergrads look into performance with the kalman filters and we did see an improvement, so if the problem is solved do let me know. Thanks for the quick response Talmo!
